github/repos.go

```diff
@@ -658,10 +630,10 @@ type DismissalRestrictions struct {
 // separate from DismissalRestrictions above because the request structure is
 // different from the response structure.
 type DismissalRestrictionsRequest struct {
-	// The list of user logins who can dismiss pull request reviews. (Required; use []string{} instead of nil for empty list.)
-	Users []string `json:"users"`
-	// The list of team slugs which can dismiss pull request reviews. (Required; use []string{} instead of nil for empty list.)
-	Teams []string `json:"teams"`
+	// The list of user logins who can dismiss pull request reviews. (Required; use nil to disable dismissal_restrictions or []string{} otherwise.)
+	Users *[]string `json:"users,omitempty"`
```

Comment:
Note that a slice is already a reference type, so having a pointer to a slice here is not necessary.
Can you please try your local patches without this `*` and only with the addition of `omitempty`?
You should get identical results.
Please report back with your findings, and if it it does indeed still work, please remove these `*`s.

Reply:
For personal repositories, the 'dismissal_restrictions' must be omitted. For repositories in an organization, the 'dismissal_restrictions' must either be an empty JSON object or JSON object with exactly two keys 'users' and 'teams'. Any other combination of repository and 'dismissal_restrictions' is rejected by the GitHub API.

If I add 'omitempty' to those fields, and also leave 'omitempty' on dismissal_restrictions, it no longer works.

Reply:
I apologize for not being clear.

I'm not asking you to change `omitempty` at all.

I'm asking you to remove two asterisks:

```go
Users []string `json:"users,omitempty"`
//...
Teams []string `json:"teams,omitempty"`
```

Reply:
Ah oops, no the error is on me. I read your comment too quickly. Take a look at these two configurations from the go playground. Here is the configuration using the pointers on the inner arrays: https://play.golang.org/p/nShoOQw1Fd_A. It allows us to specify 3 configurations. When we remove the pointers on the arrays: https://play.golang.org/p/OfPxg539U14. We can no longer generate this JSON object: `{"dismissal_restrictions":{"users":[],"teams":[]}}`

Reply:
That is good justification for why it needs to be a `*[]string` with `omitempty`. It's to enable the JSON marshaling that is needed for GitHub API.

Reply:
Ah, thank you for the playground example!  That makes sense now.